addons/mail/models/mail_channel.py

```diff
@@ -400,15 +400,26 @@ def add_members(self, partner_ids=None, guest_ids=None, invite_to_rtc_call=False
                 if post_joined_message:
                     # notify existing members with a new message in the channel
                     if member.partner_id == self.env.user.partner_id:
-                        notification = _('<div class="o_mail_notification">joined the channel</div>')
+                        notification = _(
+                            '<div class="o_mail_notification">%(member)s joined the channel</div>',
+                            member=current_partner.name or current_guest.name,
+                        )
                     else:
                         notification = _(
-                            '<div class="o_mail_notification">invited %s to the channel</div>',
-                            member.partner_id._get_html_link(),
+                            '<div class="o_mail_notification">%(current_member)s invited %(other_members)s to the channel</div>',
+                            current_member=current_partner.name or current_guest.name,
+                            other_members=member.partner_id._get_html_link(),
                         )
                     member.channel_id.message_post(body=notification, message_type="notification", subtype_xmlid="mail.mt_comment")
             for member in new_members.filtered(lambda member: member.guest_id):
-                member.channel_id.message_post(body=_('<div class="o_mail_notification">joined the channel</div>'), message_type="notification", subtype_xmlid="mail.mt_comment")
+                member.channel_id.message_post(
+                    body=_(
+                        '<div class="o_mail_notification">%(member)s joined the channel</div>',
+                        member=member.partner_id.name or member.guest_id.name,
```

Comment:
current code makes no sense for partner_id as there is a member.guest_id condition in the loop